src/rng.rs

```diff
@@ -0,0 +1,45 @@
+use crate::syscalls;
+use core::cell::Cell;
+
+const DRIVER_NUMBER: usize = 0x40001;
+
+mod command_nr {
+    pub const REQUEST_RNG: usize = 1;
+}
+
+mod subscribe_nr {
+    pub const BUFFER_FILLED: usize = 0;
+}
+
+mod allow_nr {
+    pub const SHARE_BUFFER: usize = 0;
+}
+
+pub fn fill_buffer(buf: &mut [u8]) -> bool {
```

Comment:
I'm not in love with this function name, but I don't have a much better suggestion either... `get_random`? meh...

Reply:
I'm fine merging as is, unless someone can come up with a better name...